drivers/SmartThings/matter-lock/src/new-matter-lock/init.lua

```diff
@@ -0,0 +1,1580 @@
+      end)
+    end)
+  end
+end
+
+local function require_remote_pin_handler(driver, device, ib, response)
+  if ib.data.value then
+    apply_cota_credentials_if_absent(device)
+  else
+    device:set_field(lock_utils.COTA_CRED, false, {persist = true})
+  end
+end
+
+-----------------------------------------------------
+-- Number Of Week Day Schedules Supported Per User --
+-----------------------------------------------------
+local function max_week_schedule_of_user_handler(driver, device, ib, response)
+  device:emit_event(capabilities.lockSchedules.weekDaySchedulesPerUser(ib.data.value, {visibility = {displayed = false}}))
+end
+
+-----------------------------------------------------
+-- Number Of Year Day Schedules Supported Per User --
+-----------------------------------------------------
+local function max_year_schedule_of_user_handler(driver, device, ib, response)
+  device:emit_event(capabilities.lockSchedules.yearDaySchedulesPerUser(ib.data.value, {visibility = {displayed = false}}))
+end
+
+-- Capability Handler
+-----------------
+-- Lock/Unlock --
+-----------------
+local function handle_lock(driver, device, command)
+  local ep = device:component_to_endpoint(command.component)
+  local cota_cred = device:get_field(lock_utils.COTA_CRED)
+  if cota_cred then
+    device:send(
+      DoorLock.server.commands.LockDoor(device, ep, cota_cred)
+    )
+  else
+    device:send(DoorLock.server.commands.LockDoor(device, ep))
+  end
+end
+
+local function handle_unlock(driver, device, command)
+  local ep = device:component_to_endpoint(command.component)
+  local cota_cred = device:get_field(lock_utils.COTA_CRED)
+  if cota_cred then
+    device:send(
+      DoorLock.server.commands.UnlockDoor(device, ep, cota_cred)
+    )
+  else
+    device:send(DoorLock.server.commands.UnlockDoor(device, ep))
+  end
+end
+
+----------------
+-- User Table --
+----------------
+local function add_user_to_table(device, userIdx, usrType)
+  -- Get latest user table
+  local user_table = device:get_latest_state(
+    "main",
+    capabilities.lockUsers.ID,
+    capabilities.lockUsers.users.NAME
+  ) or {}
+  local new_user_table = {}
+
+  -- Recreate user table
+  for index, entry in pairs(user_table) do
+    table.insert(new_user_table, entry)
+  end
+
+  -- Add new entry to table
+  table.insert(new_user_table, {userIndex = userIdx, userType = usrType})
+  device:emit_event(capabilities.lockUsers.users(new_user_table, {visibility = {displayed = false}}))
+end
+
+local function update_user_in_table(device, userIdx, usrType)
+  -- Get latest user table
+  local user_table = device:get_latest_state(
+    "main",
+    capabilities.lockUsers.ID,
+    capabilities.lockUsers.users.NAME
+  ) or {}
+  local new_user_table = {}
+
+  -- Recreate user table
+  local i = 0
+  for index, entry in pairs(user_table) do
+    if entry.userIndex == userIdx then
+      i = index
+    end
+    table.insert(new_user_table, entry)
+  end
```

Comment:
why are we recreating the table here? Could we not just handle it like so:
```suggestion
  local i = 0
  for index, entry in pairs(user_table) do
    if entry.userIndex == userIdx then
      i = index
      break
    end
  end
 
new_user_table = user_table
```

Where I'm not even sure we would to reinitialize `user_table` like this: `new_user_table = user_table`. Could we not just keep using `user_table`?

Reply:
I modified it. Thank you~!